### PR TITLE
[BIOMAGE-999] Add unit tests for the gem2s API point

### PR DESCRIPTION
**Pull request by robioson**
# Background
#### Link to issue
https://biomage.atlassian.net/browse/BIOMAGE-999

#### Link to staging deployment URL 
N/A

#### Links to any Pull Requests related to this
#109

#### Anything else the reviewers should know about the changes here
I initially based the tests on `pipelines.test.js` since the code for `pipelines.js` is very similar to that in `gem2s.js`. I then added tests to get 100% coverage for `gem2s.js`. It could be worthwhile coming back and doing the same for `pipelines.js`.

# Changes
### Code changes
- Created `tests/api/routes/gem2s.test.js`
- Added newly mocked function in `src/api/route-services/__mocks__/experiment.js`

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [ ] Unit tests written
- [ ] Tested locally with Inframock (with latest production data downloaded with biomage experiment pull)
- [ ] Deployed to staging

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [x] Photo of a cute animal attached to this PR

image

### CODE REVIEW

File: tests/api/routes/gem2s.test.js [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+jest.mock('sns-validator');
+jest.mock('aws-xray-sdk');
+jest.mock('../../../src/utils/authMiddlewares');
+jest.mock('../../../src/utils/logging');
+jest.mock('../../../src/cache');
+jest.mock('../../../src/api/route-services/gem2s-response');
+jest.mock('../../../src/api/general-services/pipeline-manage');
+jest.mock('../../../src/api/route-services/experiment');
+
+const basicMsg = JSON.stringify({
```
Comment (cosa65) [inline tests/api/routes/gem2s.test.js]: is this `JSON.stringify` necessary? it seems like `basicMsg` is JSON.parsed later on when it's used (except for one test)

Comment (robioson) [reply tests/api/routes/gem2s.test.js]: No, you're right, I copied it from pipeline tests so I fixed it there too.

File: tests/api/routes/pipelines.test.js [hunk 2]
```diff
@@ -110,8 +109,8 @@ describe('PipelineResults route', () => {
     expect(https.get).toHaveBeenCalledTimes(1);
   });
 
-  it('Get malformatted work results returns an error', async () => {
-    const brokenMsg = JSON.stringify();
+  it('Returns an error for malformed work', async () => {
```
Comment (robioson) [inline tests/api/routes/pipelines.test.js:112]: Just noticed - is this test actually any different to the one starting on line 63? :thinking:

Comment (cosa65) [reply tests/api/routes/pipelines.test.js:112]: Hmm brokenMsg and validMsg seem to have different properties

Comment (robioson) [reply tests/api/routes/pipelines.test.js:112]: Yes, they are invalid in different ways (`invalidMsg` is missing the `Type` field whereas `brokenMsg` is just completely empty). I will leave both as I don't want to remove a test without fully understanding the reasons both were added.

**Review by cosa65 — APPROVED** [review]

**Pull request merged**